### ISSUE TITLE
Add on scripts not working

### ISSUE THREAD

**Issue by wpmjones** [github/hubot#1001]
I'm an amateur at all of this and I just learn as I go.  Hubot and Heroku have quite the learning curve!

Raspberry Pi - Model B (running Raspbian)
Node - v0.12.1
NPM -  v2.12.0
Installed Hubot - Slack adapter
Installed Heroku
Deployed

In Slack, I can talk to Hubot and he responds as expected (help, pug me, time all work)

Installed mail, logo, wikipedia via npm (i.e. - npm install hubot-logo --save)
sudo nano external-scripts.json and added script name
rebooted Pi

Built in scripts still work.  New scripts provide no response whatsoever.

redeployed Heroku thinking that might be necessary

Same result.  No error, just no response.  I tried looking at Heroku Logs, but it doesn't seem to track script requests.  It shows a lot of ping and pong and keep alive, but nothing on scripts.

Are there other logs I can view or other ways I can determine where it's breaking?

Assumption: Slack is behaving since other scripts work.  Therefore, Heroku is likely receiving the script request and *may* even be passing the script request on to Hubot, but something is breaking in that area.

**Comment by technicalpickles** [comment]
I'm a little confused. Are you saying you are running this on a raspberry pi and heroku, or you were using it on the pi, had problems and then tried heroku to see if it's working?

> Installed mail, logo, wikipedia via npm (i.e. - npm install hubot-logo --save)
> sudo nano external-scripts.json and added script name
> rebooted Pi

You added the line like `"hubot-logo"`? The file is JSON and it's expecting an array, so you'd need to make sure there's commas after each string if you add multiple. Like:

```
[
  "hubot-help",
  "hubot-logo"
]
```

> I tried looking at Heroku Logs, but it doesn't seem to track script requests. It shows a lot of ping and pong and keep alive, but nothing on scripts.

Yeah, unfortunately hubot doesn't log anything about the scripts that are running unless the scripts use `robot.logger` themselves to do it. Once we have https://github.com/github/hubot/pull/803 I think it'd be possible to add this to help with knowing that scripts are actually being called or not.

> Are there other logs I can view or other ways I can determine where it's breaking?

`bin/hubot` logs everything to stdout. If you are running on heroku, that should ultimately make it's way to Heroku Logs for you to see. I don't know how you are launching on your pi, but it might be directing the logs elsewhere.

**Comment by wpmjones** [comment]
I was still pretty ignorant when I posted that.  I built hubot on my RPi, but now understand that pushing it to Heroku actually makes it active on the web.  I thought hubot was still hitting my Pi.  Now I know how it works and after starting from scratch about 5 times, I finally got it working properly.  

Currently working on getting hubot to talk to my home automation system and parsing the XML responses I get back from the system to tell the user what happened.  This is fun stuff!

Thanks for checking back in!
